tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py

```diff
@@ -885,23 +885,26 @@ def visit_functiondef(self, node):
         :return: None
         """
         try:
-            if node.parent.name.endswith("Client") and node.parent.name not in self.ignore_clients and node.is_method():
-                if node.name.startswith("list"):
+            if node.parent.parent.name.endswith("Client") and node.parent.parent.name not in self.ignore_clients and node.parent.is_method():
+                if node.parent.name.startswith("list"):
+                    paging_class = False
+
                     try:
-                        # infer_call_result gives the method return value as a string
-                        returns = next(node.infer_call_result()).as_string()
-                        if returns.find("ItemPaged") == -1 and returns.find("AsyncItemPaged") == -1:
-                            self.add_message(
-                                msgid="client-list-methods-use-paging", node=node, confidence=None
-                            )
+                        if "def by_page" in next(node.value.infer()).as_string():
```

Comment:
You're now passing the return directly? And is this checking that both custom paging classes, ItemPaged and AsyncItemPaged define `by_page`?